guides/hack/03-expressions-and-operators/99-yield-examples/process-file.php

```diff
@@ -9,7 +9,11 @@ function getTextFileLines(string $filename): \Generator<int, string, void> {
   }
 
   try {
-    while ($textLine = \fgets($infile)) { // while not EOF
+    while (true) {
+      $textLine = \fgets($infile);
+      if (!$textLine) {
```

Comment:
this is equivalent to the old code, but could you replace it with an explicit check for `=== false`, given falsey strings are reasonable here